src/main/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -255,6 +258,40 @@ public Object getAttribute(String key, Object defaultValue) {
         return Collections.singletonList(o);
     }
 
+    private <T> List<T> getAttributeAsList(String key, Function<Object, T> transformer) {
+        return getAttributeAsList(key).stream().map(transformer).collect(Collectors.toList());
+    }
+
+    public List<String> getAttributeAsStringList(String key, String defaultValue) {
+        return getAttributeAsList(key, x -> (x == null) ? defaultValue : String.valueOf(x));
+    }
+
+    public List<Integer> getAttributeAsIntList(String key, int defaultValue) {
+        return getAttributeAsList(key, x -> {
+                            if (x == null || x == VCFConstants.MISSING_VALUE_v4) {
+                                return defaultValue;
+                            } else if (x instanceof Integer) {
+                                return (Integer) x;
```

Comment:
`int[5]` generates an empty array with each element initialized to 0, so it will work. The same for the double.